### ISSUE TITLE
Crash on opening Honeycomb ListFragment Context Menu

### ISSUE THREAD

**Issue by ianhanniballake** [JakeWharton/ActionBarSherlock#63]
Per [this Gist example Android project code](https://gist.github.com/1220937) built against ActionBarSherlock v3.2.1 — 2011-09-12, the context menu for the ListFragment opens successfully on Gingerbread or lower devices. However, on Honeycomb devices, the following error occurs:

```
FATAL EXCEPTION: main
java.lang.ClassCastException: com.android.internal.view.menu.ContextMenuBuilder cannot be cast to com.actionbarsherlock.internal.view.menu.MenuWrapper
  at com.actionbarsherlock.internal.view.menu.MenuInflaterWrapper.inflate(MenuInflaterWrapper.java:19)
  at com.ianhanniballake.honeycombcontextmenu.ContextMenuListFragment.onCreateContextMenu(ContextMenuListFragment.java:31)
  at android.view.View.createContextMenu(View.java:5138)
  at com.android.internal.view.menu.ContextMenuBuilder.show(ContextMenuBuilder.java:81)
  ...
```

Due to miscasting the ContextMenu object to a MenuWrapper in the MenuWrapper.inflate method.

Suggested solution (which fixes this in the sample project) is to rewrite MenuWrapper.inflate to:

```
@Override
public void inflate(final int menuRes, final android.view.Menu menu) {
    if (menu instanceof ContextMenu) {
        mNativeMenuInflater.inflate(menuRes, menu);
        return;
    }
    mMenuInflater.inflate(menuRes, ((MenuWrapper) menu).unwrap());
}
```

To match the approach used in MenuInflater (which is used in the Gingerbread implementation).

**Comment by JakeWharton** [comment]
This is fixed in the dev branch. Will be released this weekend. You can
download the branch from GitHub if you need it ASAP.

---

Jake Wharton
http://about.me/jakewharton
On Sep 15, 2011 9:32 PM, "Ian Lake" <
reply@reply.github.com>
wrote:

**Comment by JakeWharton** [comment]
Released with 3.2.2. Couldn't wait...

**Comment by ianhanniballake** [comment]
Works perfectly in v3.2.2. Thanks for the quick fix.
